app/group.go

```diff
@@ -218,6 +218,21 @@ func (a *App) DeleteGroup(groupID string) (*model.Group, *model.AppError) {
 	return deletedGroup, nil
 }
 
+func (a *App) UndeleteGroup(groupID string) (*model.Group, *model.AppError) {
```

Comment:
If you decide to update the API route you can decide if you'd like to update the App method to match `(*App).RestoreChannel` and `(*App).RestoreTeam`:
```suggestion
func (a *App) RestoreGroup(groupID string) (*model.Group, *model.AppError) {
```